### PR TITLE
Add support volume scheduler hint option

### PR DESCRIPTION
**Pull request by jimmyxian**
Fix:  #827

### CODE REVIEW

File: openstack/blockstorage/extensions/schedulerhints/requests.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"regexp"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// SchedulerHints represents a set of scheduling hints that are passed to the
+// OpenStack scheduler.
+type SchedulerHints struct {
```
Comment (jtopjian) [inline openstack/blockstorage/extensions/schedulerhints/requests.go]: It looks like these fields are defined in this schema:

https://github.com/openstack/cinder/blob/b8167a5c3e5952cc52ff8844804b7a5ab36459c8/cinder/api/schemas/scheduler_hints.py

The schema defines a `query` field of type `string. Let's add that here for completeness.

Comment (jimmyxian) [reply openstack/blockstorage/extensions/schedulerhints/requests.go]: OK

File: openstack/blockstorage/extensions/schedulerhints/requests.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	// SameHost will place the volume on a back-end that hosts the given volumes.
+	SameHost []string
+
+	// LocalToInstance will place volume on same host on a given instance
+	LocalToInstance string
+
+	// Query is a conditional statement that results in back-ends able to
+	// host the volume.
+	Query []interface{}
```
Comment (jtopjian) [inline openstack/blockstorage/extensions/schedulerhints/requests.go]: Why is this `[]interface{}`?

**Comment by jtopjian** [comment]
@jimmyxian In #1243, I have enabled testing of block storage extensions in OpenLab. If you rebase this PR with the master branch and do a force push, your acceptance test will now run.

However, it is _highly_ recommended to always test your code locally and not rely on OpenLab for testing. This is outlined in our [contributor tutorial](https://github.com/gophercloud/gophercloud/blob/master/docs/contributor-tutorial/step-04-acceptance-testing.md)

**Comment by jimmyxian** [comment]
@jtopjian Rebased. 
Also, I have tested it in our environment.

File: openstack/blockstorage/extensions/schedulerhints/requests.go (hunk 2 above)
Comment (jimmyxian) [reply openstack/blockstorage/extensions/schedulerhints/requests.go]: Also, we can find `query` used in https://github.com/openstack/cinder/blob/master/cinder/scheduler/filters/json_filter.py

**Comment by jimmyxian** [comment]
recheck

**Comment by jimmyxian** [comment]
ping @jtopjian Review again. Thanks

File: openstack/blockstorage/extensions/schedulerhints/requests.go (hunk 2 above)
Comment (jtopjian) [reply openstack/blockstorage/extensions/schedulerhints/requests.go]: But it is defined as a string here: https://github.com/openstack/cinder/blob/b8167a5c3e5952cc52ff8844804b7a5ab36459c8/cinder/api/schemas/scheduler_hints.py#L69

The comments indicate that `jsonutils.loads` is used to convert a string to a dict.

**Comment by jimmyxian** [comment]
@jtopjian Review again. Thanks

File: openstack/blockstorage/extensions/schedulerhints/requests.go (hunk 2 above)
Comment (jimmyxian) [reply openstack/blockstorage/extensions/schedulerhints/requests.go]: OK,Got it. :)

**Comment by jimmyxian** [comment]
ping @jtopjian Done! Review again. Thanks

**Pull request merged**